var/spack/repos/builtin/packages/dftbplus/package.py

```diff
@@ -3,85 +3,158 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
-import os
-
 from spack.package import *
 
 
-class Dftbplus(MakefilePackage):
+class Dftbplus(CMakePackage, MakefilePackage):
     """DFTB+ is an implementation of the
     Density Functional based Tight Binding (DFTB) method,
     containing many extensions to the original method."""
 
     homepage = "https://www.dftbplus.org"
-    url = "https://github.com/dftbplus/dftbplus/archive/19.1.tar.gz"
+    url = "https://github.com/dftbplus/dftbplus/releases/download/22.1/dftbplus-22.1.tar.xz"
+    git = "https://github.com/dftbplus/dftbplus.git"
 
-    version("19.1", sha256="4d07f5c6102f06999d8cfdb1d17f5b59f9f2b804697f14b3bc562e3ea094b8a8")
+    maintainers = ["HaoZeke", "aradi", "iamashwin99"]
+    generator = "Ninja"
 
-    resource(
-        name="slakos",
-        url="https://github.com/dftbplus/testparams/archive/dftbplus-18.2.tar.gz",
-        sha256="bd191b3d240c1a81a8754a365e53a78b581fc92eb074dd5beb8b56a669a8d3d1",
-        destination="external/slakos",
-        when="@18.2:",
+    build_system(
+        conditional("cmake", when="@20.1:"),
+        conditional("makefile", when="@:19.1"),
+        default="cmake",
     )
 
-    variant("mpi", default=True, description="Build an MPI-paralelised version of the code.")
+    version("main", branch="main")
+    version("22.1", sha256="f0fc9a076aa2d7be03c31a3a845d8151fc0cc0b1d421e11c37044f78a42abb33")
+    version("21.2", sha256="e73aa698ff951b59f7fe2ea027b292bae16abf545c9fdbb11b5b5127f04a3c10")
+    version("21.1", sha256="31d5a488843a05d8589a375307a2832c1fc938f9f7d830c45a062726659e7b0a")
+    version("20.2.1", sha256="6b1827a45b20d1757119a75abcb851cd4362e7abc58601094d029ed5922d6da7")
+    version("20.2", sha256="6b1f10c7efbdaf59acf64d8fb8afc80460c0fbc6c4dbe256b760223cbd28ed2c")
+    version("20.1", sha256="a155ca927c804234587c61c4938d154f31578c816b0ce20eaee3b5d7e39d91dc")
+    version(
+        "19.1",
+        sha256="4d07f5c6102f06999d8cfdb1d17f5b59f9f2b804697f14b3bc562e3ea094b8a8",
+        deprecated=True,
+    )
 
     variant(
-        "gpu",
+        "api",
+        default=True,
+        description="Whether public API should be included and the DFTB+ library installed.",
+    )
+    variant(
+        "arpack",
         default=False,
-        description="Use the MAGMA library " "for GPU accelerated computation",
+        description="Whether the ARPACK library should be included (needed for TD-DFTB).",
+    )
+    variant(
+        "chimes",
+        default=False,
+        when="@21.2:",
+        description="Whether repulsive corrections" "via the ChIMES library should be enabled.",
     )
-
     variant(
         "elsi",
         default=False,
         description="Use the ELSI library for large scale systems. "
         "Only has any effect if you build with '+mpi'",
     )
```

Comment:
Only has any effect if you build with '+mpi', so we can make the variant also available only when mpi build is built
```suggestion
        description="Use the ELSI library for large scale systems. "
        "Only has any effect if you build with '+mpi'",
        when="+mpi"
    )
```